### PR TITLE
Nodata nan fix

### LINKED ISSUE #530: Proposal: Support loading in multiple datasets with np.nan as the nodata value.
Currently, we have run into instances where netCDF files have a nodata/_FillValue value of NaN (strange, but it happened). The current code in odc does not correctly search for the location of nodata  in the case of a NaN value (https://github.com/opendatacube/datacube-core/blob/develop/datacube/storage/storage.py#L195, and https://github.com/opendatacube/datacube-core/blob/develop/datacube/storage/storage.py#L162). It is a simple fix to use `numpy.isnan` to support this. 

We have developed a fix for this, would you be interested in a PR?

### PR DESCRIPTION
**Pull request by snowman2**
Solves loading in data with np.nan as nodata value
 - [x] Closes #530
 - [x] Tests added / passed
 - [ ] Fully documented, including `docs/about/whats_new.rst` for all changes

Note: There are other minor fixes added in as well with this PR.

### CODE REVIEW

File: datacube/storage/storage.py [hunk 1]
```diff
@@ -159,7 +159,8 @@ def read_from_source(source, dest, dst_transform, dst_nodata, dst_projection, re
                 _LOG.debug('Failed Read: %s', src, exc_info=1)
                 return
             dest = dest[offset[0]:offset[0] + tmp.shape[0], offset[1]:offset[1] + tmp.shape[1]]
-            numpy.copyto(dest, tmp, where=(tmp != src.nodata))
+            where_not_nodata = (tmp != src.nodata) if not numpy.isnan(src.nodata) else ~numpy.isnan(tmp)
```
Comment (Kirill888) [inline datacube/storage/storage.py]: double negative in a variable name is a bit confusing, how about `valid_data`/`valid_mask`?

**Review by Kirill888 — CHANGES_REQUESTED** [review]
@snowman2 can I ask you to rebase away two middle commits (changes to yaml files), just want to keep this pull request focused on one thing, by the way rasterio 1.0 had some reprojection bugs when dealing with multi-band sources, so probably want 1.0.2+

**Comment by snowman2** [comment]
@Kirill888, I renamed where_not_nodata -> where_valid_data and I removed the extra commits so the PR is clean now.

**Review by Kirill888 — APPROVED** [review]
Thanks, we'll fix up rasterio pinning separately.

**Pull request merged**